### ISSUE TITLE
ERROR: full-file-path/file.xml.gpg not found. This should not happen. Exiting...

### ISSUE THREAD

**Issue by hendercine** [StackExchange/blackbox#255]
I have just set up blackbox for the first time and have successfully added myself as admin and registered my file in blackbox. When I attempt to edit, however, I receive the error as mentioned in the title of this issue. What am I my missing?

**Comment by hendercine** [comment]
Thanks for the prompt response @tlimoncelli . I tried it all over again this morning and it worked! So I don't know what was happening. I'm certain I must have input something incorrectly but I still can't resolve where, so I've included the steps I used below anyway just for the record.

So, to answer your questions:

- I'm working in: 

Android Studio 3.1.2
Build #AI-173.4720617, built on April 13, 2018
JRE: 1.8.0_152-release-1024-b01 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.12.6 (Sierra)


- In the Android studio terminal I input this command:
`blackbox_register_new_file ~/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml`

- The output in the terminal was as follows:
```
========== PLAINFILE /Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml
========== ENCRYPTED /Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml.gpg
========== Importing keychain: START
gpg: key ****************: "My Name <my@email.com>" not changed
gpg: key ****************: "My Name <my@email.com>" not changed
gpg: Total number processed: 2
gpg:              unchanged: 2
========== Importing keychain: DONE
========== Encrypting: /Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml
========== Encrypting: DONE
========== Adding file to list.
========== CREATED: /Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml.gpg
========== UPDATING REPO:
rm 'app/src/main/res/values/file-to-encrypt.xml'
NOTE: "already tracked!" messages are safe to ignore.
[master a8c8985] registered in blackbox: /Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml
 2 files changed, 1 insertion(+)
 create mode 100644 app/src/main/res/values/file-to-encrypt.xml.gpg
========== UPDATING VCS: DONE
Local repo updated.  Please push when ready.
    git push
```

- I then input the following command and received the output below:
```
blackbox_edit_start Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml
========== PLAINFILE "Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml"
ERROR: Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml.gpg not found. This should not happen.
Exiting...
```
- I assumed I was inputting the file-path incorrectly so I tried:
```
blackbox_edit_start file-to-encrypt.xml
========== PLAINFILE "file-to-encrypt.xml"
ERROR: file-to-encrypt.xml not found in /Users/myusername/version_control_folder/project_folder/keyrings/live/blackbox-files.txt
PWD=/Users/myusername/version_control_folder/project_folder
Exiting...
```
- I then tried decrypt all, again assuming I was having path problems:

```
blackbox_decrypt_all_files
========== Importing keychain: START
gpg: key ****************: "My Name <my@email.com>" not changed
gpg: key ****************: "My Name <my@email.com>" not changed
gpg: Total number processed: 2
gpg:              unchanged: 2
========== Importing keychain: DONE
========== Decrypting new/changed files: START
gpg: can't open 'Users/myusername/version_control_folder/project_folder/app/src/main/res/values/file-to-encrypt.xml.gpg': No such file or directory
gpg: decrypt_message failed: No such file or directory
```
I then hard reset the entire work-tree back to before registering my admin account and tried it all over again with the same results. That's when I decided to post this issue.

Thanks for your time and sorry if it was wasted, because as I said up top, blackbox is now working and it's great!

**Comment by hendercine** [comment]
I think I found the issue. I encrypted another file - my google-services.json file - and when I tried to decrypt all the same error described above occurred. On a hunch, I opened at blackbox-files.txt and found that the paths were written without a leading `/`. I added the `/` to the front of each path in blackbox-files.txt and blackbox_decrypt_all_files ran successfully!

Had I the time right now (or the skills?) to contribute a fix for this I would, but I think I should leave it in more capable hands?
